### PR TITLE
Create Existing App Integration Tutorial

### PR DESCRIPTION
**Pull request by JoelMarcey**
We currently have iOS and Android existing app integration guides. I have revamped these into a single tutorial, with three sections: Objective-C, Swift and Android.

For Objective-C and and Swift, the tutorial is now based on a more real world app - integrating a React Native-based high score screen into a 2048 app.

For expediency to get the iOS stuff out, for Android, *for now*, I have kept the existing documentation (with minor updates), but am planning to try to follow the same 2048 model for it as well.

This uses the same toggler as Getting Started

> I do note the copypasta of the toggler code. I am planning another separate pull request to make that more modular and reusable across all areas of the documentation on which it seems reasonable.

screenshot 2016-05-25 15 34 27
screenshot 2016-05-25 15 40 50

### CODE REVIEW

**Comment by facebook-github-bot-7** [comment]
By analyzing the blame information on this pull request, we identified @davidyaha and @thorbenprimke to be potential reviewers.

**Comment by javache** [comment]
You mention sharing the bridge on Android (mReactInstancrHolder), but just creating a new RCTRootView every time on iOS. On iOS it is also preferable and more efficient to create the bridge once and use it for multiple root views.

File: docs/Tutorial-IntegrationWithExistingApps.md [hunk 1]
```diff
@@ -0,0 +1,745 @@
...
+
+We will add the package dependencies to a `package.json` file. Create this file in the root of your project if it does not exist.
+
+> When "root of project" is mentioned in any step, that is the location where your application's project file is located (e.g., `.xcodeproj`).
+
+Below is an example of what your `package.json` file should minimally contain.
+
+> Version numbers will vary according to your needs. Normally the latest versions for each will be sufficient.
+
+<block class="objc" />
```
Comment (janicduplessis) [inline docs/Tutorial-IntegrationWithExistingApps.md]: Maybe we can share this block with swift/objc, the only difference is the name and it doesn't really matter.

File: docs/Tutorial-IntegrationWithExistingApps.md [hunk 2]
```diff
@@ -0,0 +1,745 @@
...
+
+```bash
+{
+  "name": "NumberTileGame",
+  "version": "0.0.1",
+  "private": true,
+  "scripts": {
+  "start": "node node_modules/react-native/local-cli/cli.js start"
```
Comment (janicduplessis) [inline docs/Tutorial-IntegrationWithExistingApps.md]: Nit: indent

File: docs/Tutorial-IntegrationWithExistingApps.md [hunk 3]
```diff
@@ -0,0 +1,745 @@
...
+
+![Event Path](img/react-native-add-react-native-integration-link.png)
+
+#### Event Handler
+
+We will now add an event handler from the menu link. A method will be added to the main `ViewController` of your application. This is where `RCTRootView` comes into play.
+
+When you build a React Native application, you use the React Native packager to create an `index.ios.bundle` that will be served by the React Native server. Inside `index.ios.bundle` will be our `RNHighScore` module. So, we need to point our `RCTRootView` to the location of the `index.ios.bundle` resource (via `NSURL`) and tie it to the module.
+
+We will, for debugging purposes, log that the event handler was invoked. Then create a string with the location of our React Native code inside the `index.ios.bundle` and then create the main `RCTRootView`. Notice how we provide `HighScores` as the `moduleName` that we created [above](#the-react-native-component) when writing the code for our React Native component.
```
Comment (janicduplessis) [inline docs/Tutorial-IntegrationWithExistingApps.md]: I think it should be "... `RNHighScores` as the `moduleName` ..."

**Comment by JoelMarcey** [comment]
@javache Thanks. Yeah, that was a point @frantic made too. I wanted to get the instructions clean for a straightforward integration first. And then in a subsequent pull request we can make modifications for efficiency, etc.

However, maybe for this pull request we can add a note in the same vain as Android for now? If you have wording suggestion there, let me know; otherwise I will give it a go.

**Comment by JoelMarcey** [comment]
ugh! I come back from vacation and try to rebase this and screw everything up. Will try to fix this mess.

File: docs/Tutorial-IntegrationWithExistingApps.md (hunk 1 above)
Comment (JoelMarcey) [inline docs/Tutorial-IntegrationWithExistingApps.md]: Yeah, I thought about that -- but I wanted it to match the code on GitHub that I link to at the end as much as possible.

**Comment by JoelMarcey** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-1** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/103647103390500/int_phab) to review.

**Pull request closed without merging**